filterapi/filterconfig.go

```diff
@@ -145,13 +143,16 @@ type HeaderMatch = gwapiv1.HTTPHeaderMatch
 // besides the `Backends` field is modified to abstract the concept of a backend
 // at Envoy Gateway level to a simple name.
 type RouteRule struct {
+	// Name is the name of the route rule.
+	Name RouteRuleName `json:"name"`
 	// Headers is the list of headers to match for the routing decision.
 	// Currently, only exact match is supported.
 	Headers []HeaderMatch `json:"headers"`
-	// Backends is the list of backends to which the request should be routed to when the headers match.
```

Comment:
I think still possible with https://github.com/envoyproxy/ai-gateway/issues/588#issuecomment-2842370535 here?

Reply:
yeah it still should be possible even after this PR, just that the `backends` moved to the top level vs under Rules here.